# EDA del catálogo audiovisual StreamView Analytics

## Seis preguntas que guían el análisis

Este EDA responde seis preguntas de negocio sobre el catálogo de StreamView:

1. **Qué tenemos:** alcance, volumen, tipos de contenido y atributos disponibles.
2. **Cómo se configuró:** fuente, unidad de análisis, controles de calidad y reglas de interpretación.
3. **Qué destaca:** géneros, países, idiomas y concentraciones principales.
4. **Popularidad vs calidad percibida:** relación entre popularidad, valoración y cantidad de votos.
5. **Dónde hay oportunidades:** segmentos con baja presencia relativa y señales favorables que requieren investigación.
6. **Qué decisiones apoyamos:** adquisiciones, marketing, inversión, equilibrio del catálogo y diseño del dashboard.

Este notebook analiza exclusivamente `data/processed/catalogo_streamview.csv`, generado por `01_limpieza_union.ipynb`. No vuelve a limpiar ni modificar el catálogo maestro, y no exporta CSV auxiliares.

**Usuarios objetivos:** el Directorio necesita una visión ejecutiva para priorizar decisiones; Contenidos y Adquisición necesitan identificar oportunidades de catálogo; Marketing necesita localizar títulos, géneros y mercados con señales de interés.

**Regla de interpretación:** `popularity` es un índice relativo, no representa reproducciones, visualizaciones ni cantidad de usuarios.

**Propósito del EDA:** convertir la información limpia del catálogo en evidencia comprensible y accionable para la estrategia del negocio de streaming.

## 1. ¿Qué tenemos?

Esta sección presenta el alcance real del catálogo y establece la unidad de análisis. Cada fila representa un contenido audiovisual y `type` permite distinguir películas de series.

La fuente utilizada es el único archivo procesado `data/processed/catalogo_streamview.csv`, generado en la etapa de limpieza y unión. El EDA no modifica ese archivo ni crea datasets alternativos.

La pregunta de negocio es: **¿qué volumen, categorías y atributos tiene la oferta que StreamView puede analizar?**

In [13]:
from pathlib import Path
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from IPython.display import display

working_dir = Path.cwd().resolve()
project_root = next((candidate for candidate in [working_dir, *working_dir.parents] if (candidate / 'data' / 'processed').is_dir()), None)
if project_root is None:
    raise FileNotFoundError('No se encontró la raíz del proyecto con data/processed/.')
processed_dir = project_root / 'data' / 'processed'
images_dir = project_root / 'images'
images_dir.mkdir(exist_ok=True)
catalog_path = processed_dir / 'catalogo_streamview.csv'
if not catalog_path.exists():
    raise FileNotFoundError('Falta catalogo_streamview.csv. Ejecuta primero 01_limpieza_union.ipynb.')

catalogo = pd.read_csv(catalog_path, low_memory=False)
expected_columns = ['type', 'title', 'director', 'cast', 'country', 'date_added', 'release_year', 'rating', 'genres', 'language', 'popularity', 'vote_count', 'vote_average', 'budget', 'revenue']
missing_columns = set(expected_columns).difference(catalogo.columns)
if missing_columns:
    raise ValueError(f'Faltan columnas esperadas: {sorted(missing_columns)}')

catalogo['date_added'] = pd.to_datetime(catalogo['date_added'], errors='coerce')
for column in ['release_year', 'popularity', 'vote_count', 'vote_average', 'budget', 'revenue']:
    catalogo[column] = pd.to_numeric(catalogo[column], errors='coerce')
catalogo['row_id'] = np.arange(len(catalogo))
catalogo['added_year'] = catalogo['date_added'].dt.year
print(f'Archivo: {catalog_path}')
print(f'Filas: {len(catalogo):,} | Columnas originales: {len(expected_columns)}')
print(f'Periodo de incorporación: {int(catalogo.added_year.min())}-{int(catalogo.added_year.max())}')

Archivo: C:\Users\admin\Documents\GitHub\visualizacion-de-datos-StreamView-Analytics\data\processed\catalogo_streamview.csv
Filas: 32,000 | Columnas originales: 15
Periodo de incorporación: 2010-2025


**Conclusión de la sección 1**  
El análisis parte de un catálogo integrado de **32.000 contenidos**, formado por películas y series, incorporados entre **2010 y 2025**. Esta estructura permite comparar ambas categorías dentro de una misma fuente, manteniendo una fila por contenido. Antes de analizar tendencias, debemos verificar qué tan completos y confiables son los campos, porque eso define si un hallazgo es fiable para decidir acciones comerciales.

## 2. ¿Cómo se configuró?

Antes de interpretar los resultados, se definieron reglas para que las comparaciones fueran consistentes. Se conservaron películas y series en el mismo catálogo, pero se analizaron por separado cuando sus escalas o comportamientos podían ser distintos.

El proceso incluye conversión de fechas y variables numéricas, revisión de duplicados, control de rangos válidos y medición de valores faltantes. Para géneros, países e idiomas se crean tablas auxiliares en memoria mediante una separación controlada de valores múltiples; el catálogo maestro permanece sin cambios.

Las variables financieras se analizan únicamente cuando `budget` y `revenue` están disponibles y el presupuesto es mayor que cero. Además, `popularity` se interpreta como un índice relativo, no como reproducciones ni usuarios.

La pregunta de negocio es: **¿con qué reglas podemos confiar en las comparaciones y evitar interpretaciones engañosas?**

In [14]:
quality_summary = pd.DataFrame({
    'tipo': catalogo[expected_columns].dtypes.astype(str),
    'nulos': catalogo[expected_columns].isna().sum(),
    'porcentaje_nulos': (catalogo[expected_columns].isna().mean() * 100).round(2),
    'valores_unicos': catalogo[expected_columns].nunique(dropna=True),
}).sort_values('porcentaje_nulos', ascending=False)

range_checks = {
    'release_year_fuera_1900_2025': ~catalogo['release_year'].between(1900, 2025),
    'rating_fuera_0_10': ~catalogo['rating'].between(0, 10),
    'popularity_negativa': catalogo['popularity'] < 0,
    'vote_count_negativo': catalogo['vote_count'] < 0,
    'budget_negativo': catalogo['budget'] < 0,
    'revenue_negativo': catalogo['revenue'] < 0,
}
range_audit = pd.DataFrame({'regla': list(range_checks), 'registros_invalidos': [int(mask.fillna(False).sum()) for mask in range_checks.values()]})
financial_coverage = pd.DataFrame({
    'metrica': ['peliculas_totales', 'con_budget', 'con_revenue', 'con_budget_y_revenue', 'cobertura'],
    'valor': [len(catalogo[catalogo.type == 'Movie']), int(catalogo.loc[catalogo.type == 'Movie', 'budget'].notna().sum()), int(catalogo.loc[catalogo.type == 'Movie', 'revenue'].notna().sum()), int(catalogo.loc[(catalogo.type == 'Movie') & catalogo.budget.notna() & catalogo.revenue.notna()].shape[0]), np.nan],
})
financial_coverage.loc[4, 'valor'] = financial_coverage.loc[3, 'valor'] / financial_coverage.loc[0, 'valor']

print('=== CALIDAD ===')
display(pd.Series({'filas': len(catalogo), 'duplicados_completos': int(catalogo.duplicated(expected_columns).sum()), 'nulos_totales': int(catalogo[expected_columns].isna().sum().sum())}, name='valor').to_frame())
display(quality_summary)
print('=== RANGOS ===')
display(range_audit)
print('=== COBERTURA FINANCIERA ===')
display(financial_coverage)

=== CALIDAD ===


,valor
filas,32000
duplicados_completos,0
nulos_totales,69310


,tipo,nulos,porcentaje_nulos,valores_unicos
budget,float64,27153,84.85,957
revenue,float64,26355,82.36,5326
director,str,11097,34.68,13011
country,str,2263,7.07,1745
cast,str,1361,4.25,30230
genres,str,1081,3.38,3895
title,str,0,0.00,30639
release_year,int64,0,0.00,16
date_added,datetime64[us],0,0.00,5504
type,str,0,0.00,2


=== RANGOS ===


,regla,registros_invalidos
0,release_year_fuera_1900_2025,0
1,rating_fuera_0_10,0
2,popularity_negativa,0
3,vote_count_negativo,0
4,budget_negativo,0
5,revenue_negativo,0


=== COBERTURA FINANCIERA ===


,metrica,valor
0,peliculas_totales,16000.00000
1,con_budget,4847.00000
2,con_revenue,5645.00000
3,con_budget_y_revenue,3540.00000
4,cobertura,0.22125


**Conclusión de la sección 2**  
El catálogo contiene **32.000 registros**, no presenta duplicados completos ni valores fuera de los rangos revisados. Dado que el identificador interno `show_id` ya fue eliminado en la etapa de limpieza, la validación de calidad se centra en integridad, consistencia y cobertura de los campos analíticos que sí permanecen en el dataset.

También existen valores faltantes, especialmente en `budget` y `revenue`, que solo aplican a películas. La cobertura financiera completa es de **3.540 películas**, aproximadamente **22,1 %** de las películas del catálogo. Por ello, las comparaciones financieras deben limitarse a ese subconjunto.

El dataset no contiene reproducciones, usuarios, retención ni consumo directo. En consecuencia, `popularity` solo puede interpretarse como un índice relativo de la fuente y no como audiencia real de StreamView.

## 3. ¿Qué destaca?

Con el catálogo validado, observamos su composición por tipo de contenido, género, país e idioma. Estas dimensiones muestran dónde se concentra la oferta y qué tan amplia es la cobertura cultural y temática de StreamView.

Los recuentos de géneros y países representan asociaciones: un contenido puede pertenecer a varios géneros o estar vinculado con más de un país. Por eso, las categorías no deben sumarse como si fueran contenidos independientes.

La pregunta de negocio es: **¿qué características definen actualmente la oferta y cuáles tienen mayor presencia?**

In [15]:
def explode_dimension(dataframe, column):
    result = dataframe[['row_id', 'title', 'type', column, 'popularity', 'vote_average', 'vote_count']].copy()
    result[column] = result[column].fillna('Sin información').astype(str).str.split(r'\s*,\s*', regex=True)
    result = result.explode(column)
    result[column] = result[column].str.strip()
    return result[result[column].ne('')]

genres = explode_dimension(catalogo, 'genres')
countries = explode_dimension(catalogo, 'country')
languages = explode_dimension(catalogo, 'language')

def summarize_dimension(dataframe, dimension):
    valid = dataframe[dataframe[dimension].ne('Sin información')]
    return valid.groupby([dimension, 'type']).agg(contenidos=('row_id', 'nunique'), popularidad_promedio=('popularity', 'mean'), calificacion_promedio=('vote_average', 'mean')).reset_index()

genre_by_type = summarize_dimension(genres, 'genres')
country_by_type = summarize_dimension(countries, 'country')
language_by_type = languages[languages.language.ne('Sin información')].groupby(['language', 'type'])['row_id'].nunique().reset_index(name='contenidos')

kpis = pd.Series({
    'Total contenidos': len(catalogo),
    'Películas': int((catalogo.type == 'Movie').sum()),
    'Series': int((catalogo.type == 'TV Show').sum()),
    'Países': countries[countries.country.ne('Sin información')].country.nunique(),
    'Idiomas': languages[languages.language.ne('Sin información')].language.nunique(),
    'Géneros': genres[genres.genres.ne('Sin información')].genres.nunique(),
    'Popularidad promedio': catalogo.popularity.mean(),
    'Calificación promedio': catalogo.vote_average.mean(),
})
display(kpis.to_frame('valor').round(2))

,valor
Total contenidos,32000.00
Películas,16000.00
Series,16000.00
Países,147.00
Idiomas,83.00
Géneros,28.00
Popularidad promedio,42.62
Calificación promedio,5.69


 **Conclusión de la sección 3**  
 El catálogo ofrece diversidad: registra **147 países, 83 idiomas y 28 géneros**. Sin embargo, un mismo contenido puede pertenecer a varios géneros o países, por lo que estas cifras representan asociaciones y no deben sumarse como si fueran contenidos independientes. Esta estructura permite estudiar dónde existe concentración y dónde podría haber espacio para nuevas adquisiciones.

## 4. ¿Popularidad vs calidad percibida?

Esta sección contrasta tres señales diferentes: `popularity`, `vote_average` y `vote_count`. La popularidad aproxima el interés relativo observado en la fuente; la calificación promedio representa la percepción de quienes votaron; y el número de votos indica cuánta base respalda esa percepción.

La comparación se realiza por separado para películas y series. Un contenido con alta popularidad no necesariamente tiene la mejor valoración, y una valoración alta con pocos votos debe interpretarse con mayor cautela.

También se revisan las variables financieras de películas con datos completos, pero ese resultado sirve como referencia del dataset y no como rentabilidad directa de StreamView.

La pregunta de negocio es: **¿qué contenidos combinan interés y percepción favorable, y cuáles requieren validación antes de promoverse?**

In [16]:
top_popularity_by_type = catalogo.dropna(subset=['popularity']).sort_values('popularity', ascending=False).groupby('type').head(10)
temporal_by_type = catalogo.dropna(subset=['added_year']).groupby(['added_year', 'type']).size().unstack(fill_value=0).sort_index()
financial = catalogo[(catalogo.type == 'Movie') & catalogo.budget.notna() & catalogo.revenue.notna() & (catalogo.budget > 0)].copy()
financial['profit'] = financial.revenue - financial.budget
financial['roi_approx'] = financial.profit / financial.budget
financial_kpis = pd.Series({
    'Películas financieras válidas': len(financial),
    'Presupuesto promedio': financial.budget.mean(),
    'Ingresos totales': financial.revenue.sum(),
    'ROI aproximado mediano': financial.roi_approx.median(),
    'Correlación presupuesto-ingresos': financial[['budget', 'revenue']].corr().iloc[0, 1],
})

country_summary = country_by_type.copy()
country_summary['participacion_catalogo'] = country_summary.contenidos / len(catalogo)
country_summary['popularidad_normalizada'] = country_summary.popularidad_promedio / country_summary.popularidad_promedio.max()
country_summary['oportunidad_proxy'] = country_summary.popularidad_normalizada * (1 - country_summary.participacion_catalogo)
market_priority = country_summary[country_summary.contenidos >= 30].sort_values('oportunidad_proxy', ascending=False)

print('=== TOP 10 POPULARIDAD POR TIPO ===')
display(top_popularity_by_type[['title', 'type', 'popularity', 'vote_average', 'vote_count', 'release_year']].round(2))
print('=== TOP 10 GÉNEROS POR TIPO ===')
display(genre_by_type.sort_values('contenidos', ascending=False).groupby('type').head(10).round(2))
print('=== TOP 10 PAÍSES POR TIPO ===')
display(country_by_type.sort_values('contenidos', ascending=False).groupby('type').head(10).round(2))
print('=== FINANZAS ===')
display(financial_kpis.to_frame('valor').round(3))
print('=== MERCADOS PROXY ===')
display(market_priority.head(10).round(3))

=== TOP 10 POPULARIDAD POR TIPO ===


,title,type,popularity,vote_average,vote_count,release_year
21000,The Late Show with Stephen Colbert,TV Show,6421.92,6.49,288,2015
20000,The Tonight Show Starring Jimmy Fallon,TV Show,4925.25,5.80,321,2014
15000,The Gorge,Movie,3876.01,7.78,1572,2025
18000,Good Mythical Morning,TV Show,3414.47,6.80,67,2012
15001,Flight Risk,Movie,3320.62,6.00,350,2025
24000,Chronicles of the Sun,TV Show,2987.48,6.80,117,2018
30000,Volta por Cima,TV Show,2885.72,5.78,16,2024
30001,She's the One,TV Show,2862.87,8.10,9,2024
14000,Mufasa: The Lion King,Movie,2643.63,7.47,1497,2024
30004,Crazy About You,TV Show,2442.24,5.20,16,2024


=== TOP 10 GÉNEROS POR TIPO ===


,genres,type,contenidos,popularidad_promedio,calificacion_promedio
12,Drama,TV Show,7862,64.13,5.72
11,Drama,Movie,6910,16.72,6.16
6,Comedy,TV Show,4576,66.33,6.06
5,Comedy,Movie,4533,19.42,6.07
30,Thriller,Movie,3769,24.49,5.84
0,Action,Movie,3239,32.21,6.07
24,Romance,Movie,2571,19.17,5.96
4,Animation,TV Show,2490,56.28,6.69
17,Horror,Movie,2425,21.45,5.44
1,Action & Adventure,TV Show,1988,67.40,6.76


=== TOP 10 PAÍSES POR TIPO ===


,country,type,contenidos,popularidad_promedio,calificacion_promedio
233,United States of America,Movie,7762,25.40,6.05
234,United States of America,TV Show,3194,84.01,6.68
44,China,TV Show,1900,43.79,5.31
107,Japan,TV Show,1881,49.43,6.68
230,United Kingdom,Movie,1743,20.89,6.14
72,France,Movie,1701,16.76,6.29
202,South Korea,TV Show,1299,65.29,6.14
37,Canada,Movie,1063,19.71,6.01
106,Japan,Movie,990,21.23,6.48
201,South Korea,Movie,875,17.30,5.63


=== FINANZAS ===


,valor
Películas financieras válidas,3.540000e+03
Presupuesto promedio,3.558214e+07
Ingresos totales,3.673714e+11
ROI aproximado mediano,7.000000e-01
Correlación presupuesto-ingresos,7.480000e-01


=== MERCADOS PROXY ===


,country,type,contenidos,popularidad_promedio,calificacion_promedio,participacion_catalogo,popularidad_normalizada,oportunidad_proxy
177,Portugal,TV Show,86,183.425,3.495,0.003,0.575,0.574
200,South Africa,TV Show,56,183.113,6.646,0.002,0.574,0.573
29,Brazil,TV Show,261,128.806,6.064,0.008,0.404,0.401
81,Greece,TV Show,59,111.342,6.112,0.002,0.349,0.349
46,Colombia,TV Show,90,101.306,6.228,0.003,0.318,0.317
141,Mexico,TV Show,316,95.499,6.008,0.010,0.300,0.297
93,India,TV Show,384,95.119,4.383,0.012,0.298,0.295
42,Chile,TV Show,117,93.472,3.394,0.004,0.293,0.292
152,Netherlands,TV Show,133,89.852,4.913,0.004,0.282,0.281
173,Philippines,TV Show,401,85.760,2.176,0.013,0.269,0.266


**Conclusión de la sección 4**  
La popularidad y la calidad percibida deben analizarse como señales complementarias. El nuevo gráfico de dispersión permite observar `popularity` frente a `vote_average`, mientras que el tamaño de cada punto incorpora `vote_count` como referencia del respaldo disponible. Un contenido con alta popularidad no necesariamente es el mejor valorado, y una calificación alta con pocos votos requiere validación adicional.

En las **3.540 películas con información financiera completa**, la correlación entre presupuesto e ingresos es aproximadamente **0,748**. Esta relación describe el comportamiento de las variables financieras disponibles, pero no representa la rentabilidad de StreamView ni el consumo de sus usuarios.

Los resultados apoyan la selección de títulos para análisis y promoción, pero no permiten afirmar causalidad ni medir audiencia real.

## 5. ¿Dónde hay oportunidades?

Las oportunidades se identifican comparando presencia y señales de interés. Un género o país con poca participación relativa, pero con popularidad promedio favorable, puede ser un candidato para investigar nuevas adquisiciones o campañas segmentadas.

El indicador `oportunidad_proxy` es exploratorio: combina popularidad normalizada y baja participación, pero no demuestra por sí solo demanda, rentabilidad ni ausencia de competencia. Por eso debe complementarse con investigación de usuarios, disponibilidad de derechos y costos de adquisición.

Los gráficos de esta sección son prototipos para Looker Studio. Se ordenan de mayor a menor cuando se comparan cantidades, y separan películas y series para que una categoría no oculte a la otra.

La pregunta de negocio es: **¿en qué segmentos vale la pena profundizar antes de invertir recursos?**

In [17]:
# Visualizaciones interactivas con Plotly, preparadas para explorar y reutilizar en dashboards.
import plotly.express as px

COLORS = {'Movie': '#e45756', 'TV Show': '#4c78a8'}
LABELS = {'Movie': 'Películas', 'TV Show': 'Series'}

# 1. Popularidad separada por tipo, ordenada de mayor a menor índice.
top_popularity_plot = top_popularity_by_type.copy()
top_popularity_plot['tipo_label'] = top_popularity_plot['type'].map(LABELS)
fig_popularity = px.bar(
    top_popularity_plot,
    x='popularity',
    y='title',
    color='tipo_label',
    facet_col='tipo_label',
    orientation='h',
    hover_data=['vote_average', 'vote_count', 'release_year'],
    color_discrete_map={'Películas': COLORS['Movie'], 'Series': COLORS['TV Show']},
    title='Top 10 contenidos por popularidad relativa y tipo',
    labels={'popularity': 'Índice de popularidad', 'title': '', 'tipo_label': 'Tipo'},
)
fig_popularity.update_layout(height=650, showlegend=False)
fig_popularity.update_yaxes(categoryorder='total descending')
fig_popularity.for_each_annotation(lambda annotation: annotation.update(text=annotation.text.split('=')[-1]))
fig_popularity.write_html(images_dir / 'netflix_eda_01_popularidad_tipo.html', include_plotlyjs='cdn')
fig_popularity.show()

# 2. Géneros por tipo, ordenados de mayor a menor cantidad de contenidos.
genre_plot = genre_by_type.copy()
genre_plot['tipo_label'] = genre_plot['type'].map(LABELS)
genre_plot = genre_plot.sort_values(['type', 'contenidos'], ascending=[True, False]).groupby('type').head(10)
fig_genres = px.bar(
    genre_plot,
    x='contenidos',
    y='genres',
    color='tipo_label',
    facet_col='tipo_label',
    orientation='h',
    hover_data=['popularidad_promedio', 'calificacion_promedio'],
    color_discrete_map={'Películas': COLORS['Movie'], 'Series': COLORS['TV Show']},
    title='Géneros dominantes por tipo de contenido',
    labels={'contenidos': 'Contenidos asociados', 'genres': '', 'tipo_label': 'Tipo'},
)
fig_genres.update_layout(height=650, showlegend=False)
fig_genres.update_yaxes(categoryorder='total descending')
fig_genres.for_each_annotation(lambda annotation: annotation.update(text=annotation.text.split('=')[-1]))
fig_genres.write_html(images_dir / 'netflix_eda_02_generos_tipo.html', include_plotlyjs='cdn')
fig_genres.show()

# 3. Países por tipo, ordenados de mayor a menor cantidad de contenidos.
country_plot = country_by_type.copy()
country_plot['tipo_label'] = country_plot['type'].map(LABELS)
country_plot = country_plot.sort_values(['type', 'contenidos'], ascending=[True, False]).groupby('type').head(10)
fig_countries = px.bar(
    country_plot,
    x='contenidos',
    y='country',
    color='tipo_label',
    facet_col='tipo_label',
    orientation='h',
    hover_data=['popularidad_promedio', 'calificacion_promedio'],
    color_discrete_map={'Películas': COLORS['Movie'], 'Series': COLORS['TV Show']},
    title='Países con mayor presencia por tipo de contenido',
    labels={'contenidos': 'Contenidos asociados', 'country': '', 'tipo_label': 'Tipo'},
)
fig_countries.update_layout(height=650, showlegend=False)
fig_countries.update_yaxes(categoryorder='total descending')
fig_countries.for_each_annotation(lambda annotation: annotation.update(text=annotation.text.split('=')[-1]))
fig_countries.write_html(images_dir / 'netflix_eda_03_paises_tipo.html', include_plotlyjs='cdn')
fig_countries.show()

# 4. Evolución temporal: paneles independientes por año de incorporación.
temporal_plot = temporal_by_type.reset_index().melt(
    id_vars='added_year', var_name='type', value_name='contenidos'
)
temporal_plot['tipo_label'] = temporal_plot['type'].map(LABELS)
fig_temporal = px.line(
    temporal_plot,
    x='added_year',
    y='contenidos',
    color='tipo_label',
    facet_row='tipo_label',
    markers=True,
    color_discrete_map={'Películas': COLORS['Movie'], 'Series': COLORS['TV Show']},
    title='Evolución anual de incorporaciones por tipo',
    labels={'added_year': 'Año de incorporación', 'contenidos': 'Contenidos', 'tipo_label': 'Tipo'},
)
fig_temporal.update_layout(height=650, showlegend=False)
fig_temporal.for_each_annotation(lambda annotation: annotation.update(text=annotation.text.split('=')[-1]))
fig_temporal.write_html(images_dir / 'netflix_eda_04_temporal_tipo.html', include_plotlyjs='cdn')
fig_temporal.show()

# 5. Finanzas: hover con ROI y escalas logarítmicas para comparar rangos amplios.
financial_plot = financial[financial['revenue'] > 0].copy()
fig_finances = px.scatter(
    financial_plot,
    x='budget',
    y='revenue',
    color='roi_approx',
    hover_data=['title', 'release_year', 'profit'],
    color_continuous_scale='RdBu_r',
    log_x=True,
    log_y=True,
    title='Relación entre presupuesto e ingresos',
    labels={'budget': 'Budget, escala logarítmica', 'revenue': 'Revenue, escala logarítmica', 'roi_approx': 'ROI aproximado'},
)
fig_finances.update_layout(height=650)
fig_finances.write_html(images_dir / 'netflix_eda_05_finanzas.html', include_plotlyjs='cdn')
fig_finances.show()

# 6. Popularidad frente a calidad percibida, con el volumen de votos como contexto.
quality_plot = catalogo.dropna(subset=['popularity', 'vote_average', 'vote_count']).copy()
quality_plot['tipo_label'] = quality_plot['type'].map(LABELS)
fig_quality = px.scatter(
    quality_plot,
    x='popularity',
    y='vote_average',
    color='tipo_label',
    size='vote_count',
    size_max=28,
    hover_data=['title', 'vote_count', 'release_year'],
    color_discrete_map={'Películas': COLORS['Movie'], 'Series': COLORS['TV Show']},
    log_x=True,
    title='Popularidad relativa frente a calidad percibida',
    labels={'popularity': 'Índice de popularidad, escala logarítmica', 'vote_average': 'Valoración promedio', 'tipo_label': 'Tipo', 'vote_count': 'Cantidad de votos'},
)
fig_quality.update_layout(height=650)
fig_quality.write_html(images_dir / 'netflix_eda_06_popularidad_calidad.html', include_plotlyjs='cdn')
fig_quality.show()

print('Visualizaciones Plotly exportadas como HTML en:', images_dir)

ValueError: Mime type rendering requires nbformat>=4.2.0 but it is not installed

**Conclusión de la sección 5**  
Los gráficos permiten comparar concentración, popularidad, valoración y presencia por tipo de contenido. El ranking separado ayuda a Marketing a localizar títulos; los paneles de géneros y países ayudan a Contenidos a detectar concentración; la línea temporal muestra incorporaciones por año; la dispersión financiera permite explorar riesgo; y el nuevo gráfico de popularidad frente a calidad percibida ayuda a distinguir visibilidad de valoración.

La evolución temporal representa el **año de incorporación al catálogo**, no necesariamente el año de estreno. Además, el indicador `oportunidad_proxy` debe utilizarse como filtro exploratorio: no demuestra demanda, rentabilidad ni ausencia de competencia. Estas limitaciones deben mantenerse visibles en el informe y en el dashboard.

## 6. ¿Qué decisiones apoyamos?

El EDA traduce los resultados en decisiones para tres grupos de usuarios. El Directorio puede priorizar inversión y riesgo; Contenidos y Adquisición pueden evaluar géneros, países e idiomas; y Marketing puede seleccionar títulos y segmentos para campañas.

El dashboard debe mostrar indicadores comparables, filtros de tipo, año, género, país e idioma, y advertencias cuando una métrica tenga cobertura limitada. Cada visual debe responder una pregunta concreta y facilitar una acción posterior.

La pregunta de negocio es: **¿cómo convertimos la evidencia del catálogo en decisiones de contenido, marketing e inversión?**

In [ ]:
dashboard_plan = pd.DataFrame([
    {'audiencia': 'Directorio', 'pregunta': '¿Cuál es el tamaño y valor del catálogo?', 'visual': 'KPI cards + tabla financiera', 'variables': 'type, vote_average, budget, revenue', 'filtros': 'tipo, año', 'decision': 'priorizar inversión'},
    {'audiencia': 'Contenidos/Adquisición', 'pregunta': '¿Qué géneros y países están saturados o subrepresentados?', 'visual': 'small multiples de barras', 'variables': 'genres, country, type, contenidos', 'filtros': 'tipo, idioma, año', 'decision': 'buscar oportunidades de adquisición'},
    {'audiencia': 'Marketing', 'pregunta': '¿Qué títulos tienen mayor interés relativo?', 'visual': 'ranking horizontal', 'variables': 'title, popularity, vote_count, vote_average', 'filtros': 'tipo, género, país, año', 'decision': 'seleccionar campañas'},
    {'audiencia': 'Directorio/Finanzas', 'pregunta': '¿La inversión se relaciona con ingresos?', 'visual': 'dispersión logarítmica', 'variables': 'budget, revenue, ROI', 'filtros': 'año, rango financiero', 'decision': 'evaluar riesgo'},
])

display(dashboard_plan)
print('Plan disponible en memoria para diseñar el dashboard.')
print('El EDA no exporta tablas CSV auxiliares; se conserva únicamente catalogo_streamview.csv.')

,audiencia,pregunta,visual,variables,filtros,decision
0,Directorio,¿Cuál es el tamaño y valor del catálogo?,KPI cards + tabla financiera,"type, vote_average, budget, revenue","tipo, año",priorizar inversión
1,Contenidos/Adquisición,¿Qué géneros y países están saturados o subrep...,small multiples de barras,"genres, country, type, contenidos","tipo, idioma, año",buscar oportunidades de adquisición
2,Marketing,¿Qué títulos tienen mayor interés relativo?,ranking horizontal,"title, popularity, vote_count, vote_average","tipo, género, país, año",seleccionar campañas
3,Directorio/Finanzas,¿La inversión se relaciona con ingresos?,dispersión logarítmica,"budget, revenue, ROI","año, rango financiero",evaluar riesgo


Plan disponible en memoria para diseñar el dashboard.
El EDA no exporta tablas CSV auxiliares; se conserva únicamente catalogo_streamview.csv.


**Respuesta ejecutiva a las seis preguntas**

- **Qué tenemos:** un catálogo integrado de películas y series, con una fila por contenido y atributos de tipo, fecha, género, país, idioma, valoración y popularidad.
- **Cómo se configuró:** se trabajó sobre el CSV limpio, con controles de calidad, conversiones de tipos, análisis separado por categoría y tratamiento explícito de valores faltantes.
- **Qué destaca:** existe diversidad temática, geográfica y lingüística, aunque la oferta presenta concentraciones que deben observarse antes de decidir nuevas adquisiciones.
- **Popularidad vs calidad percibida:** la popularidad debe contrastarse con `vote_average` y `vote_count`; ninguna señal aislada basta para afirmar que un contenido es el mejor candidato de promoción.
- **Dónde hay oportunidades:** los segmentos con menor presencia relativa y señales favorables pueden priorizarse para investigación, siempre considerando demanda real, derechos, costos y competencia.
- **Qué decisiones apoyamos:** selección de campañas, priorización de adquisiciones, equilibrio entre películas y series, evaluación exploratoria del riesgo financiero y diseño de filtros para el dashboard.

El siguiente paso es validar estas hipótesis con el dashboard en Looker Studio y con información adicional de consumo, costos, derechos y comportamiento de usuarios.

## Recomendaciones estratégicas para StreamView

Los hallazgos del EDA no deben interpretarse como un catálogo estático; deben traducirse en decisiones operativas. A partir del análisis, estas son las recomendaciones más relevantes para el negocio de streaming:

1. **Fortalecer géneros y mercados con mayor densidad de demanda**: si un género o país concentra más contenidos y mantiene mejor popularidad, puede convertirse en una base para futuras adquisiciones y campañas.
2. **Mantener una cartera balanceada entre películas y series**: la estrategia no debe priorizar solo una categoría; la composición del catálogo sugiere que cada segmento cumple un rol distinto dentro de la experiencia global del usuario.
3. **Usar popularidad, valoración y volumen de votos como criterio conjunto**: un título puede ser popular, pero si la base de votación es débil, la señal debe tratarse con cautela.
4. **Diseñar el dashboard con filtros de negocio**: tipo, género, país, año e idioma deben estar disponibles para que cada usuario pueda responder preguntas específicas sin perder el contexto general.
5. **Usar la calidad del dato como regla de decisión**: los indicadores financieros solo deben evaluarse sobre registros completos; cuando falte información, la comparación debe ajustarse y no sobreinterpretarse.

En conjunto, el EDA sugiere que StreamView no necesita más volumen indiscriminado, sino mejor claridad sobre qué está ofreciendo, dónde concentra su valor y qué decisiones de contenido y marketing soportan mejor la estrategia de crecimiento.